lib/plugins/input/journald-upload.js

```diff
@@ -167,20 +173,25 @@ JournaldUpload.prototype.journaldHttpHandler = function (req, res) {
       res.end(`invalid logs token in url ${req.url}`)
       return
     }
-
-    req.on('data', function (data) {
-      bodyIn += String(data)
-    })
-
-    req.on('end', function endHandler () {
+    
+    req.pipe(createStreamThrottle(this.throughputPerClient)).pipe(split()).on('data', function emitLine (data) {
+      // console.log(data, data.indexOf('\n'))
+      // logs are spearated by empty lines
+      if (data.length == 0 && bodyIn.length > 0) {
+        self.parseBody(bodyIn, token)
+        bodyIn = ''
+        return
+      } 
+      bodyIn += `${data}\n`  
+    }).on('end', function endHandler () {
```

Comment:
How does this work?

This part looks strange. :thinking: 
```js
self.parseBody(bodyIn, token)
bodyIn = ''
```

Reply:
An empty line `data.length == 0` indicates the next log entry. The journald export format sends many log entries with N fields.  

```
field=value
field2=value
...
\n
field=value
feild1=valeu
...
\n
```

parseBody just parses one log entry with N fields, like 

```
field=value
field2=value
```
Parsing small chunks and emitting them to pipeline allows the streaming mode. 
Before, we waited for the complete upload, which could result in out-of-memory errros.

Reply:
Ahaaa, okay. Got it! :smile: